tests/unittests/test_parent_child_sync.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+from unittest import mock
+from parameterized import parameterized
+from tap_intercom.client import IntercomClient
+from tap_intercom.streams import Conversations
+from tap_intercom.sync import sync
+import singer
+from test_conversation_part_bookmarks import Catalog
+
+# Mock function for transform
+def transform(*args, **kwargs):
+    # Return record itself
+    return args[0]
+
+class TestParentChildSync(unittest.TestCase):
+    """
+        Test cases to verify the parent-child behaviour
+    """
+
+    @parameterized.expand([
+        ['parent_child_selected', ['conversations', 'conversation_parts'], {'stream': 'conversations'}],
+        ['parent_selected', ['conversations'], {'stream': 'conversations'}],
+        ['child_selected', ['conversation_parts'], {'stream': 'conversations'}]
+    ])
+    @mock.patch('tap_intercom.streams.IncrementalStream.sync', return_value = {})
+    def test_parent_child_selection(self, name, test_data, expected_data, mocked_sync):
+        """
+            Test case to verify we add parent stream to sync if
+            child is selected and parent stream is not selected
+        """
+        sync({}, {}, Catalog(test_data))
+        args, kwargs = mocked_sync.call_args
+        # verify the 'sync' is called with parent stream
+        self.assertEqual(args[1], expected_data)
+
+    @parameterized.expand([
+        ['parent_child_selected_start_date', [['conversations', 'conversation_parts'], {}], '2021-01-01'],
+        ['parent_child_selected_state', [['conversations', 'conversation_parts'], {'bookmarks': {'conversation_parts': {'updated_at': '2021-01-02'}, \
+            'conversations': {'updated_at': '2021-01-03'}}}], '2021-01-02'],
+        ['parent_selected_start_date', [['conversations'], {}], '2021-01-01'],
+        ['parent_selected_state', [['conversations'], {'bookmarks': {'conversation_parts': {'updated_at': '2021-01-02'}, \
+            'conversations': {'updated_at': '2021-01-03'}}}], '2021-01-03'],
+        ['child_selected_start_date', [['conversation_parts'], {}], '2021-01-01'],
+        ['child_selected_state', [['conversation_parts'], {'bookmarks': {'conversation_parts': {'updated_at': '2021-01-02'}, \
+            'conversations': {'updated_at': '2021-01-03'}}}], '2021-01-02'],
+    ])
+    @mock.patch('singer.write_schema')
+    @mock.patch('tap_intercom.streams.Conversations.get_records')
+    def test_parent_child_records_sync(self, name, test_data, expected_data, mocked_get_records, mocked_write_schema):
+        """
+            Test case to verify we set expected start date to sync for parent-child streams
+        """
+        client = IntercomClient('test_access_token', 300)
+        conversations = Conversations(client, Catalog(test_data[0]), test_data[0])
+        state = test_data[1]
+        config = {'start_date': '2021-01-01'}
+        conversations.sync(state, {}, {}, config, None)
```

Comment:
```suggestion
        conversations.sync(state=state, stream_schema={}, stream_metadata={}, config=config, transformer=None)
```

This will give better readability.

Reply:
Updated the code.

Reply:
I can't see the changes been made.